### PR TITLE
Add support for LDAP tls

### LINKED ISSUE #150: [Feature request] Support TLS for LDAP
TLS support for LDAP could be very useful.

For example Google Workspace provides Secure LDAP that allows to read group membership of Google Workspace throw LDAP protocol. But unfortunately it requires TLS configuration as far as I understood.

I take a look in this project code and realised that the code relies on ldap3 library which support TLS. So it seems it should be possible to add TLS support for LDAP without big code changes.

### PR DESCRIPTION
**Pull request by dm3ch**
/fix #150

### CODE REVIEW

**Comment by dm3ch** [comment]
I haven't yet have a chance to test it. When I would test it on my setup I would mark this PR as ready

**Comment by dm3ch** [comment]
I just realised that I wouldn't be able to test it.
Cause I was going to test it with Google Secure LDAP, but I just found that Secure LDAP is only available for Buisnes Plus plan, while I have only Buisnes Standart plan, so I forced to use Google API in my case

**Comment by primetheus** [comment]
@dm3ch thanks for opening this up! I'll setup a local environment and test it to make sure it works

File: .env.example.ldap [hunk 1]
```diff
@@ -53,6 +53,19 @@ LDAP_BIND_PASSWORD="password"
...
+## Path to private key file. Optional.
+# LDAP_SSL_PRIVATE_KEY=private.key
+## Path to server certificate file. Optional.
+# LDAP_SSL_CERTIFICATE=cert.pem
+## Validate server cert. Optional, requires cert by default.
+# LDAP_SSL_VALIDATE=CERT_REQUIRED
+## Used SSL version. Optional, uses maximum supported version by default.
+# LDAP_SSL_VERSION=PROTOCOL_TLS
+## CA certs path. Optional, if doesn't specified system CA used.
+# LDAP_SSL_CA_CERTS=cacert.b64
```
Comment (primetheus) [inline .env.example.ldap:67]:
```suggestion
## Use ssl. Optional, disabled by default.
LDAP_USE_SSL=true
## Path to private key file. Optional.
LDAP_SSL_PRIVATE_KEY=private.key
## Path to server certificate file. Optional.
LDAP_SSL_CERTIFICATE=cert.pem
## Validate server cert. Optional, requires cert by default.
LDAP_SSL_VALIDATE=CERT_REQUIRED
## Used SSL version. Optional, uses maximum supported version by default.
LDAP_SSL_VERSION=PROTOCOL_TLS
## CA certs path. Optional, if doesn't specified system CA used.
LDAP_SSL_CA_CERTS=cacert.b64
```

**Comment by lb-cake** [comment]
hey @primetheus, do mind giving this a re-review? if you want help testing on a local environment, I wouldn't mind helping you out with that.

**Comment by primetheus** [comment]
yes, apologies for the late reply. I'll have this wrapped up tonight!

**Review by primetheus — APPROVED** [review]

**Pull request merged**